state/peerAccountsDB.go

```diff
@@ -60,44 +43,20 @@ func (adb *PeerAccountsDB) MarkSnapshotDone() {
 	handleLoggingWhenError("error while putting active DB value into main storer", err)
 }
 
-func (adb *PeerAccountsDB) getTrieStorageManagerAndLatestEpoch() (common.StorageManager, uint32, error) {
-	trieStorageManager := adb.mainTrie.GetStorageManager()
-	epoch, err := trieStorageManager.GetLatestStorageEpoch()
-	if err != nil {
-		return nil, 0, fmt.Errorf("%w while getting the latest storage epoch", err)
-	}
-
-	return trieStorageManager, epoch, nil
-}
-
 // SnapshotState triggers the snapshotting process of the state trie
 func (adb *PeerAccountsDB) SnapshotState(rootHash []byte) {
-	log.Trace("peerAccountsDB.SnapshotState", "root hash", rootHash)
-	trieStorageManager, epoch, err := adb.getTrieStorageManagerAndLatestEpoch()
-	if err != nil {
-		log.Error("SnapshotState error", "err", err.Error())
-		return
-	}
+	adb.mutOp.Lock()
+	defer adb.mutOp.Unlock()
 
-	if !trieStorageManager.ShouldTakeSnapshot() {
-		log.Debug("skipping snapshot for rootHash", "hash", rootHash)
+	trieStorageManager, epoch, shouldTakeSnapshot := adb.prepareSnapshot(rootHash)
+	if !shouldTakeSnapshot {
```

Comment:
Perhaps log a message for not taking the snapshot?

Reply:
It is logged inside `prepareSnapshot`